graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -357,12 +357,21 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
                                                      inner_location, context)
         basic_blocks.extend(inner_basic_blocks)
 
+    marked_location_stack = context['marked_location_stack']
```

Comment:
Rather than grabbing the reference like this, and counting on the reader to remember that `marked_location_stack` is actually part of the context, let's try to use `context['marked_location_stack']` directly each time unless there's very good reason not to. They technically do the same thing, but we want it to be impossible for the reader to miss the fact that the context is being read or mutated.